app/services/saml_request_validator.rb

```diff
@@ -41,14 +42,21 @@ def authorized_authn_context
   end
 
   def valid_authn_context?
+    valid_contexts = Saml::Idp::Constants::VALID_AUTHN_CONTEXTS
```

Comment:
This should be a `dup` so we don't accidentally try to mutate a shared constant. `+=` does not mutate luckily... 

```suggestion
    valid_contexts = Saml::Idp::Constants::VALID_AUTHN_CONTEXTS.dup
```

Reply:
Whew good catch, thanks!